packages/wrangler/src/api/integrations/bindings/index.ts

```diff
@@ -0,0 +1,151 @@
+import { Miniflare } from "miniflare";
+import { readConfig } from "../../../config";
+import { getBindings } from "../../../dev";
+import { getBoundRegisteredWorkers } from "../../../dev-registry";
+import { buildMiniflareBindingOptions } from "../../../dev/miniflare";
+import { getServiceBindings } from "./services";
+import type { MiniflareOptions } from "miniflare";
+
+/**
+ * Options for the `getBindingsProxy` utility
+ */
+export type GetBindingsProxyOptions = {
+	/**
+	 * The path to the config object to use (default `wrangler.toml`)
+	 */
+	configPath?: string;
+	/**
+	 * Flag to indicate the utility to read a json config file (`wrangler.json`)
+	 * instead of the toml one (`wrangler.toml`)
+	 *
+	 * Note: this feature is experimental
+	 */
+	experimentalJsonConfig?: boolean;
+	/**
+	 * Indicates if and where to persist the bindings data, if not present or `true` it defaults to the same location
+	 * used by wrangler v3: `.wrangler/state/v3` (so that the same data can be easily used by the caller and wrangler).
+	 * If `false` is specified no data is persisted on the filesystem.
+	 */
+	persist?: boolean | { path: string };
+};
+
+/**
+ * Result of the `getBindingsProxy` utility
+ */
+export type BindingsProxy<Bindings = Record<string, unknown>> = {
+	/**
+	 * Object containing the various proxies
+	 */
+	bindings: Bindings;
+	/**
+	 * Function used to dispose of the child process providing the bindings implementation
+	 */
+	dispose: () => Promise<void>;
+};
+
+/**
+ * By reading from a `wrangler.toml` file this function generates proxy binding objects that can be
+ * used to simulate the interaction with bindings during local development in a Node.js environment
+ *
+ * @param options The various options that can tweak this function's behavior
+ * @returns An Object containing the generated proxies alongside other related utilities
+ */
+export async function getBindingsProxy<Bindings = Record<string, unknown>>(
+	options: GetBindingsProxyOptions = {}
+): Promise<BindingsProxy<Bindings>> {
+	const miniflareOptions = await getMiniflareOptionsFromConfig(options);
+
+	const mf = new Miniflare({
+		script: "",
+		modules: true,
+		...(miniflareOptions as Record<string, unknown>),
+	});
+
+	const bindings: Bindings = await mf.getBindings();
+
+	return {
+		bindings,
+		dispose: () => mf.dispose(),
+	};
+}
+
+async function getMiniflareOptionsFromConfig(
+	opts: GetBindingsProxyOptions
+): Promise<Partial<MiniflareOptions>> {
+	const rawConfig = readConfig(opts.configPath, {
+		experimentalJsonConfig: opts.experimentalJsonConfig,
+	});
+
+	// getBindingsProxy doesn't currently support selecting an environment
+	const env = undefined;
```

Comment:
Yes I had `env` part of the `GetBindingsProxyOptions` type before, it can definitely be added but I figured not to get bagged too much into this right now, as a first iteration, if you're ok with it, I'd go with `getBindingsProxy` always just using the top level environment, we can then always support selecting different environments later on 🙂